python/plugins/processing/algs/qgis/Datasources2Vrt.py

```diff
@@ -82,151 +81,44 @@ def defaultFileExtension(self):
-                    fieldDef = layerDef.GetFieldDefn(fieldIdx)
-                    vrt += '<Field name="{}" type="{}"'.format(self.XmlEsc(fieldDef.GetName()), self.fieldType2Name(fieldDef.GetType()))
-                    if not schema:
-                        vrt += ' src="{}"'.format(self.XmlEsc(fieldDef.GetName()))
-                    if fieldDef.GetWidth() > 0:
-                        vrt += ' width="{}"'.format(fieldDef.GetWidth())
-                    if fieldDef.GetPrecision() > 0:
-                        vrt += ' precision="{}"'.format(fieldDef.GetPrecision())
-                    vrt += '/>'
-
-                vrt += '</OGRVRTLayer>'
-
-            srcDS.Destroy()
-
-        if union:
+        if unioned:
             vrt += '</OGRVRTUnionLayer>'
-
         vrt += '</OGRVRTDataSource>'
 
-        #TODO: pretty-print XML
-
-        if outFile is not None:
-            with codecs.open(outFile, 'w') as f:
-                f.write(vrt)
-
-        return vrt
-
-    def GeomType2Name(self, geomType):
-        if geomType == ogr.wkbUnknown:
-            return 'wkbUnknown'
-        elif geomType == ogr.wkbPoint:
-            return 'wkbPoint'
-        elif geomType == ogr.wkbLineString:
-            return 'wkbLineString'
-        elif geomType == ogr.wkbPolygon:
-            return 'wkbPolygon'
-        elif geomType == ogr.wkbMultiPoint:
-            return 'wkbMultiPoint'
-        elif geomType == ogr.wkbMultiLineString:
-            return 'wkbMultiLineString'
-        elif geomType == ogr.wkbMultiPolygon:
-            return 'wkbMultiPolygon'
-        elif geomType == ogr.wkbGeometryCollection:
-            return 'wkbGeometryCollection'
-        elif geomType == ogr.wkbNone:
-            return 'wkbNone'
-        elif geomType == ogr.wkbLinearRing:
-            return 'wkbLinearRing'
-        else:
-            return 'wkbUnknown'
-
-    def fieldType2Name(self, fieldType):
-        if fieldType == ogr.OFTInteger:
-            return 'Integer'
-        elif fieldType == ogr.OFTString:
-            return 'String'
-        elif fieldType == ogr.OFTReal:
-            return 'Real'
-        elif fieldType == ogr.OFTStringList:
-            return 'StringList'
-        elif fieldType == ogr.OFTIntegerList:
-            return 'IntegerList'
-        elif fieldType == ogr.OFTRealList:
-            return 'RealList'
-        elif fieldType == ogr.OFTBinary:
-            return 'Binary'
-        elif fieldType == ogr.OFTDate:
-            return 'Date'
-        elif fieldType == ogr.OFTTime:
-            return 'Time'
-        elif fieldType == ogr.OFTDateTime:
-            return 'DateTime'
-        else:
-            return 'String'
-
-    def XmlEsc(self, text):
-        return xml.sax.saxutils.escape(text)
+        with open(vrtPath, 'w', encoding='utf-8') as f:
+            f.write(vrt)
+
+        return {self.OUTPUT: vrtPath, self.VRT_STRING: vrt}
+
+    def escape(self, text):
```

Comment:
Could this be replaced by QString::toHtmlEscaped()?

Reply:
If I'm not wrong, we can not use `QString` in python now, it is mapped to `str`.

Reply:
Ok, what about the python html.escape method?